include/range/v3/iterator_range.hpp

```diff
@@ -35,45 +35,48 @@ namespace ranges
         /// @{
         template<typename I, typename S /*= I*/>
         struct iterator_range
-          : tagged_pair<tag::begin(I), tag::end(S)>
+          : tagged_compressed_tuple<tag::begin(I), tag::end(S)>
           , view_interface<iterator_range<I, S>>
         {
         private:
-            using base_t_ = tagged_pair<tag::begin(I), tag::end(S)>;
+            using base_t = tagged_compressed_tuple<tag::begin(I), tag::end(S)>;
         public:
             using iterator = I;
             using sentinel = S;
         #ifndef RANGES_DOXYGEN_INVOKED
             using const_iterator = I; // Mostly to avoid spurious errors in Boost.Range
         #endif
 
+            using base_t::begin;
+            using base_t::end;
+
             iterator_range() = default;
             constexpr iterator_range(I begin, S end)
-              : base_t_{detail::move(begin), detail::move(end)}
+              : base_t{detail::move(begin), detail::move(end)}
             {}
             template<typename X, typename Y,
-                CONCEPT_REQUIRES_(Constructible<I, X &&>() && Constructible<S, Y &&>())>
+                CONCEPT_REQUIRES_(Constructible<I, X>() && Constructible<S, Y>())>
```

Comment:
`std::is_constructible<T, Args...>` and `ranges::Constructible<T, Args...>` are defined in terms of `std::declval<Args>()`,  which is an lvalue when `Args` is an lvalue reference type and an xvalue otherwise. The `&&` has no effect.

The rule holds for `std::experimental::ranges::Constructible` as well, albeit with a slightly different proof.